### PR TITLE
502 user should verify his changed email or mobile

### PR DESCRIPTION
**Pull request by yaroslav-fedyshyn-nordwhale**
This PR is for #502 (https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/502)

- Add endpoint to send verification code to new mobile number
- add endpoint to send verification code to new email address
- add endpoint to check verification code and save new phone number in private db
- add endpoint to check verification code and save new email address in private db

Link to GoodDapp's PR - https://github.com/GoodDollar/GoodDAPP/pull/680

### CODE REVIEW

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> point 2.
> we dont need duplicate endpoint code for re-verifaction.
> in the if condition you showed me check if email/mobile is different than current saved.

Done. See new commits

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> 1. we have a bug, also in the regular signup flow. A user can send a different mobile/email then the one he verified. to fix it:
> 
> * in sendotp, sendemail save the mobile+email to user otp record
> * in verify mobile/email copy the email/mobile form the otp record to the user private record (dont take it from "body")
> * in signup /storage/adduser - dont take mobile/email from incoming user record unless we are in dev env
> 
> 1. we dont need separate code for signup verificaiton and re-verification. use same code.

Done

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> point 2.
> we dont need duplicate endpoint code for re-verifaction.
> in the if condition you showed me check if email/mobile is different than current saved.

Done

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -325,30 +342,38 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
           log.debug('send new user email validation code', code)
         }
+
         // updates/adds user with the emailVerificationCode to be used for verification later and with mauticId
         storage.updateUser({
           identifier: user.loggedInAs,
           mauticId: userRec.mauticId,
-          emailVerificationCode: code
+          emailVerificationCode: code,
+          otp: { email }
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: wouldn't this overwrite the current otp record values?

Comment (yaroslav-fedyshyn-nordwhale) [reply src/server/verification/verificationAPI.js]: @sirpy, no, it doesn't. The gun DB is merging incoming data, not replacing the existing objects with the new one.

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> you still didnt remove all new duplicate endpoints like sendnewotp

done

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
Here is the Dapp PR for sending email/mobile code fixes.https://github.com/GoodDollar/GoodDAPP/pull/714

File: src/server/verification/verificationAPI.js (hunk 1 above)
Comment (sirpy) [reply src/server/verification/verificationAPI.js]: its not gundb is mongodb used for the private user database

Comment (yaroslav-fedyshyn-nordwhale) [reply src/server/verification/verificationAPI.js]: @sirpy done. see new commit

**Pull request merged**